cmd/collect-profiles/main.go

```diff
@@ -330,30 +330,21 @@ func requestURLBody(httpClient *http.Client, u *url.URL) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
-func populateServingCert(ctx context.Context, client client.Client) (*tls.Certificate, error) {
+func populateServingCert(ctx context.Context, client client.Client) error {
 	secret := &corev1.Secret{}
 	err := client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: pprofSecretName}, secret)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	certPEMBytes, privKeyPEMBytes, err := generateCertAndKey()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	secret.Data[corev1.TLSCertKey] = certPEMBytes
 	secret.Data[corev1.TLSPrivateKeyKey] = privKeyPEMBytes
-
-	if err = client.Update(ctx, secret); err != nil {
-		return nil, err
-	}
-	// Create tlsCert for client use
-	tlsCert, err := tls.X509KeyPair(certPEMBytes, privKeyPEMBytes)
-	if err != nil {
-		return nil, err
-	}
-	return &tlsCert, nil
+	return client.Update(ctx, secret)
```

Comment:
Hi @m1kola , thanks! Yeah, I didn't find the `updateTimeStamp` field either. Not sure why. However, can we recreate it instead of `update` it?  I'm not sure if a lot of folks care about this `creationTimestamp` field, but it's a little confusing. Anyway, looks good to me.

Reply:
@jianzhangbjz `creationTimestamp`  and `deletionTimestamp` are fields which are part of metadata which we don't explicitly set (every Kubernetes object has these). Update timestamp doesn't exist in the object metadata.

I don't think deleting and re-creating the secret is an option because the OLM deployment depends on it. When we update the secret - it is being updated (eventually) in the OLM deployment volume.

Another way to debug this - you can follow logs and look for ["generating a new cert and key"](https://github.com/openshift/operator-framework-olm/pull/884/files#diff-d1735878151d4ab08ece1912bc0940b39cf994d6b958fed3ae91e08a6c6092ddR151) line which should be logged only when we re-generate the cert.

Reply:
Yes, thanks!